### ISSUE TITLE
option to disable "All Mail"?

### ISSUE THREAD

**Issue by mipak** [ProtonMail/proton-bridge#152]
I run into similar issues as reported in #63 (that I get spam in All Mail, while any other provider does not include spam in All Mail) and would like to just disable the "All Mail" folder, so my IMAP client doesn't even see it.

Is there currently any option to do that without forking the code and exclude the folder? 
Could be a simple cmdline parameter that would fixes such cases quickly.

thx

**Comment by andrzejsza** [comment]
hi @mipak . there is no option to do that at the moment, but we are looking into this (for internal reference - GODT-759). We'll get back to you once we have more work done there.

**Comment by mipak** [comment]
just did a quick test by modifying `/internal/store/mailbox_counts.go`.

removed line 103 

`- {pmapi.AllMailLabel, "All Mail", "#000", -5, true, 0, 0},`

and added pmapi.AllMailLabel to 112

```
- case pmapi.StarredLabel, pmapi.AllSentLabel, pmapi.AllDraftsLabel:
+ case pmapi.StarredLabel, pmapi.AllSentLabel, pmapi.AllDraftsLabel, pmapi.AllMailLabel:
```

just needs to be a config parameter like `"allmail": false` which shouldn't break anything for anybody

**Comment by tseaver** [comment]
Please allow this feature, even if enabled only as a config file setting -- the "All Mail" feature may be useful in the browser client, but it is actively harmful for me in the IMAP bridge.

**Comment by prymu** [comment]
> Please allow this feature, even if enabled only as a config file setting -- the "All Mail" feature may be useful in the browser client, but it is actively harmful for me in the IMAP bridge.

second this

**Comment by andrzejsza** [comment]
yes, it will happen. the plan is to have an easy toggle in advanced settings.

**Comment by mipak** [comment]
for some wired and unsolvable reason my email client eM Client doesn't fetch the Inbox and only "All Mail" on the proton bridge - works perfect on any other imap servers or gmail. Once it fetched "All Mail" the email also appear in Inbox. 

When I apply my patch (above) which hides "All Mail", I don't get any new emails any more at all. Tried to debug but without success and "eM Client" support couldn't help either.

So either my change isn't a "clean solution" or something else is strange. I know "eM Client" is no supported client, so could be a strange bug with them in combination with proton bridge in general.

**Comment by tseaver** [comment]
FTR, the changes for this issue landed in commits:

- https://github.com/ProtonMail/proton-bridge/commit/76469969f399b8ba97f636652ba8112523a6f787
- https://github.com/ProtonMail/proton-bridge/commit/8b39ea4acb1857265820a42817a27c3bcdfaad4f
